include/swift/AST/Expr.h

```diff
@@ -1841,71 +1832,35 @@ class DynamicSubscriptExpr final
 /// member, which is to be resolved with context sensitive type information into
 /// bar.foo.  These always have unresolved type.
 class UnresolvedMemberExpr final
-    : public Expr,
-      public TrailingCallArguments<UnresolvedMemberExpr> {
+    : public Expr {
   SourceLoc DotLoc;
   DeclNameLoc NameLoc;
   DeclNameRef Name;
-  Expr *Argument;
-
-  UnresolvedMemberExpr(SourceLoc dotLoc, DeclNameLoc nameLoc,
-                       DeclNameRef name, Expr *argument,
-                       ArrayRef<Identifier> argLabels,
-                       ArrayRef<SourceLoc> argLabelLocs,
-                       bool hasTrailingClosure,
-                       bool implicit);
 
 public:
-  /// Create a new unresolved member expression with no arguments.
-  static UnresolvedMemberExpr *create(ASTContext &ctx, SourceLoc dotLoc,
-                                      DeclNameLoc nameLoc, DeclNameRef name,
-                                      bool implicit);
-
-  /// Create a new unresolved member expression.
-  static UnresolvedMemberExpr *create(ASTContext &ctx, SourceLoc dotLoc,
-                                      DeclNameLoc nameLoc, DeclNameRef name,
-                                      SourceLoc lParenLoc,
-                                      ArrayRef<Expr *> args,
-                                      ArrayRef<Identifier> argLabels,
-                                      ArrayRef<SourceLoc> argLabelLocs,
-                                      SourceLoc rParenLoc,
-                                      ArrayRef<TrailingClosure> trailingClosures,
-                                      bool implicit);
+  UnresolvedMemberExpr(SourceLoc dotLoc, DeclNameLoc nameLoc, DeclNameRef name,
+                       bool implicit)
+    : Expr(ExprKind::UnresolvedMember, implicit), DotLoc(dotLoc),
+      NameLoc(nameLoc), Name(name) {}
```

Comment:
You'll want to initialise `Bits.UnresolvedMemberExpr.FunctionRefKind` to `FunctionRefKind::Unapplied` here, as the bitfield doesn't guarantee zero-initialisation IIRC. 

Ideally we'd also use `FunctionRefKind::Compound` if `nameLoc` is a compound name, but unfortunately I believe that would be source breaking for the following:

```swift
struct S {
  static func foo(_ x: Int) -> S! { S() }
}

let x: S = .foo(_:)(5)
```

as previously we (incorrectly) classified this as a `DoubleApply`, which allows IUOs, but classifying as `Compound` currently forbids IUOs.

The proper solution will be to redesign `FunctionRefKind` to store compound-ness as a separate bit, distinct from the apply level, which would allow for IUO use with compound names. For now though, would you mind adding a FIXME here?